configs/Phi-3-mini-4k-instruct/finetuning/olive-config.json

```diff
@@ -0,0 +1,93 @@
+                "task": "text-generation",
+                "from_pretrained_args": {
+                    "trust_remote_code": true
+                }
+            }
+        }
+    },
+    "systems": {
+        "local_system": {
+            "type": "LocalSystem",
+            "config": {
+                "accelerators": [
+                    {
+                        "device": "gpu"
+                    }
+                ]
+            }
+        }
+    },
+    "data_configs": {
+        "dataset_default_train": {
+            "name": "dataset_default",
+            "type": "HuggingfaceContainer",
+            "params_config": {
+#data_configs_data_files_extension_start
+            <!--    "data_name": "<data_configs_data_files_extension>", -->
+#data_configs_data_files_extension_end
+                "<data_files>":"<data_configs_data_files>",
+                "split": "<data_configs_split>",
+                "component_kwargs": {
+                    "pre_process_data": {
+                        "dataset_type": "<dataset_type>",
+                        "text_cols": <text_cols>,
+                        "text_template": "<text_template>",
+                        "corpus_strategy": "<line_by_line>",
+                        "source_max_len": <source_max_len>,
+                        "pad_to_max_len": <pad_to_max_len>,
+                        "use_attention_mask": false
+                    }
+                }
+            }
+        }
+    },
+    "passes": {
+        "qlora": {
+            "type": "QLoRA",
+            "config": {
+                "compute_dtype": "<compute_dtype>",
+                "quant_type": "<quant_type>",
+                "double_quant": <double_quant>,
+                "lora_r": <lora_r>,
+                "lora_alpha": <lora_alpha>,
+                "lora_dropout": <lora_dropout>,
+                "train_data_config": "dataset_default_train",
+                "eval_dataset_size": <eval_dataset_size>,
+                "training_args": {
+                    "seed": <training_args_seed>,
+                    "data_seed": <training_args_data_seed>,
+                    "per_device_train_batch_size": <per_device_train_batch_size>,
+                    "per_device_eval_batch_size": <per_device_eval_batch_size>,
+                    "gradient_accumulation_steps": <gradient_accumulation_steps>,
+                    "gradient_checkpointing": <gradient_checkpointing>,
+                    "learning_rate": <learning_rate>,
+                    "num_train_epochs":<num_train_epochs>,
+                    "max_steps": <max_steps>,
+                    "logging_steps": 10,
+                    "evaluation_strategy": "steps",
+                    "eval_steps": 187,
+                    "group_by_length": true,
+                    "adam_beta2": 0.999,
+                    "max_grad_norm": 0.3,
+                    "output_dir": "<output_dir>",
+                    "report_to": "wandb"
```

Comment:
@vriveras Will `wandb` be a default hard requirement?

Reply:
Good point no. We should only provide how to do it.